woocommerce/woocommerce
# CYS - AI Offline flow

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #41623  

This pull request addresses situations in which the AI is offline on the intro page or when making a request to one of the AI endpoints.



### How to test the changes in this Pull Request:

1. Create a new site either via local environment or JN. Make sure to install WooCommerce Beta Tester as well.
2. Connect your site with Jetpack
3. Enable `customize-store` feature flag in `Tools -> WCA Test Helper -> Features`
4. Install and activate  the nightly build of WooCommerce Blocks https://github.com/woocommerce/woocommerce-blocks/releases/download/nightly/woocommerce-blocks-trunk-nightly.zip
5. Go to `WooCommerce -> Home` and click on `Customize your store` task.

#### AI is online when the user lands on the intro page

1. Click on `Design with AI` button and proceed to the assembler hub.
2. Confirm call to `/wc/private/ai/patterns` and response is `{"ai_content_generated":true}`.

#### AI is offline when the user lands on the intro page

1.  Go to `WooCommerce -> Home` and click on `Customize your store` task.
2. Open your browser's inspector and use Chrome's content override [feature](https://developer.chrome.com/docs/devtools/overrides/) to change the content of https://status.openai.com/api/v2/status.json call. Change `status.indicator` to `major`. 
3. Perform a hard refresh
4. Click on `Design with AI` button and proceed to the assembler hub.
5.  Confirm call to `/wc/private/ai/patterns` and response is `removed: true` in the network tab (inspector)
6. The Welcome modal should have the updated copy Mrk6SERPZ4KrFHSjM0a8TK-fi-7573%3A50973



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Tweaks for the CYS when the A.I is offline.

#### Comment 

</details>
